Blog with Core Features

shah-tejas opened the pull request:
Fixes #24 
Implements core features of the Blog design.

Following features need to be implemented:
- Implement Search Dropdown
- Handle and add onClick to "This Month Campaign" on the right sidebar
- Add Relevant Articles at the end of a Blog Post
- Add commenting/replying feature to the Blog post.

JElgar commented on src/components/app_download_square/AppDownloadSquare.module.scss:
```diff
@@ -0,0 +1,71 @@
+    text-align: center;
+    margin: 10px 50px;
+    
+    & div {
+        padding-top: 20px;
+    }
+
+    & p {
+        margin: 0px;
+    }
```
```suggestion
    div {
        padding-top: 20px;
    }

    p {
        margin: 0px;
    }
```

shah-tejas commented:
Hi @JElgar,

I have addressed your review comments as follows:

> But hopefully, you can just merge my change for that
Added the `react-markdown` package from your commit. Looks awesome, and was pretty straightforward!

> 1. I'm being really anti the whole `& a {}` thing atm (I've done an example comment somewhere but it's really up to you if you care/think I'm wrong
Sounds good, I have made the changes to remove `&` for all the nesting in the scss files. Hopefully, I removed them all.

> 2. I think sass always benefits from some more nesting, from my perspective the more the merrier (obviously without repeating code)
Addressed.

> 3. Have you duplicated the svgs for the download the app (apple-store.svg, google-store.svg)?
I think I just made the svg code in 1 line. It reduced the file size by a lot, so I think that would be a good optimization. But if we happen to change this svg's frequently, it should be okay to keep it indented. So git shows us what changes are being made to the code.

> 4. I spotted this, Im guessing the image being broke is me doing something dumb during upload. But I think the titles should cut off if they get too long? Maybe have max 2/3 lines whatever looks best then just use text-overflow ellipsis
Good eye! I added an ellipses for this.

> 5. Can we add some gradient over this so the text is more readable? (Or a black div or whatever you know what I mean)
I was not sure how to address this. I tried to add a blur effect to the image, and added a shadow to the text. It looks more readable now, but different than the original design. Please let me know if you'd like to change it?

Please let me know if any other thing needs to be changed. I will continue working on the other pending features for the blog.

Thanks!

JElgar commented:
Sorry I've taken so long to reply to this @shah-tejas! That's all perfect thank you! I'll take a look at those images, I think we usually just have a something% black thing over the image but if that doesn't work I'm sure we can work something out. 

My thinking @shah-tejas and @stellenberger is we get this in asap as I think it's looking great! I suggest we:

> Handle and add onClick to "This Month Campaign" on the right sidebar 
Do this

> Implement Search Dropdown
If it's easy enough can we just hide this for now (the search bar) and then add this in after we've released the blog?

Everything else (other than the images stuff which I'm happy to take a look at) I think we can hold off on for now!

Does that sound reasonable? Am I missing anything important?

JElgar commented on src/components/avatar/Avatar.module.scss:
```diff
@@ -0,0 +1,23 @@
+@import '../../utils/variables';
+
+.avatarContainer {
+    width: 60px;
+    height: 60px;
+    background: transparent;
+
+    & > img {
+        width: 100%;
+        height: 100%;
+        border-radius: 50%;
+    }
+    & > i {
+        font-size: 60px;
+    }
+}
+
+@media(max-width: $small) {
+    .avatarContainer {
+        width: 24px;
+        height: 24px;
+    }
+}
```
Oh one style thing!

I think we've been going for nesting media queries as well! So like this:

```suggestion
.avatarContainer {
    width: 60px;
    height: 60px;
    background: transparent;

    & > img {
        width: 100%;
        height: 100%;
        border-radius: 50%;
    }
    & > i {
        font-size: 60px;
    }
    
    @media(max-width: $small) {
        width: 24px;
        height: 24px;
    }
}
```

This just keeps all the styling for `.avatarContainer` in the same place but with the tradeoff of having to write more of these `@media(max-width: $small)`

JElgar commented on src/components/download_buttons/DownloadButtons.jsx:
```diff
@@ -1,14 +1,15 @@
         <img src={appleStore} alt="app download from apple store" />
       </a>
-      <a href="https://play.google.com/store/apps/details?id=com.nowu.app" target="_blank" rel="noopener noreferrer">
+      <a href={googleStoreURL} target="_blank" rel="noopener noreferrer">
```
😍

tunangst commented:
Hello there! Sorry I could only do a brief scroll over before going to work, and pardon if you already fixed this.
I believe JElgar has SASS full nesting in the components (@JElgar correct me if I am wrong, but I believe you got me on this a few times).
What I mean by this is everything stemming from the main section container.

I can make some suggestions about what I mean hopefully tomorrow, but I hope you know what I mean :].

shah-tejas commented:
Hi @JElgar @tunangst,

As discussed, I have commented out the <BlogSearchControls/> and the <MonthCampaigns/>
Please let me know if everything else is okay, to merge this PR.

Thanks,
Tejas.

tunangst commented on src/domain/blog/components/MonthCampaigns.module.scss:
```diff
@@ -0,0 +1,19 @@
+@import '../../../utils/variables';
+
+.container {
+    width: 100%;
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    align-items: center;
+    > * {
+        margin-bottom: 20px;
+    }
+}
+
+.title {
+    font-weight: $bold;
+    font-size: 30px;
+    line-height: 34px;
+    color: $black;
+}
```
```suggestion
.container {
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    > * {
        margin-bottom: 20px;
    }
    .title {
        font-weight: $bold;
        font-size: 30px;
        line-height: 34px;
        color: $black;
    }
}
```

JElgar commented on src/domain/blog/blog_article/BlogArticle.module.scss:
```diff
@@ -0,0 +1,248 @@
+            margin-right: 5px;
+            margin-bottom: 15px;
+            font-size: 27px;
+        }
+    }
+
+    .blogSubtitle {
+        & * {
+            font-size: 17px;
+            line-height: 21px;
+        }
+    }
+
+    .textSectionTitle, .textSectionContent p {
+        font-size: 17px;
+    }
+    
+    .authorInfo {
+        width: 95%;
+        padding: 50px 10px;
+        flex-direction: column;
+        justify-content: flex-start;
+
+        .author {
+            width: 100%;
+            margin: 0;
+
+            .authorName {
+                font-size: 17px;
+                margin-left: 10px;
+            }
+        }
+
+        .authorDescription {
+            width: 100%;
+            font-weight: $regular;
+            font-size: 14px;
+        }
+    }
+}
```
@tunangst you hero! Happy with this @shah-tejas?

shah-tejas replied:
Yes looks good. I will make these changes.

tunangst commented:
I like to nest my queries together at the end of the file also, but unless the leads 👍 this formatting idea change, we have to style it this way too.

Overall runs great on my side, I have not run across any errors or dev issues. I do have some styling ideas that may conflict with what the designers say, but I will post them in the next comment just in case.

JElgar commented:
Ok, I propose a plan! Let's get what on dev now into staging and give it a test (mainly the modal service that probably needs testing). Once we are happy with what in staging let's deploy that and then get this (the blog) into staging. From there we can test it with the actual blog content (which hopefully will start being uploaded for realz tomorrow morning). @shah-tejas @stellenberger @tunangst does that sound alright?

tunangst commented:
![Screenshot (16)](https://user-images.githubusercontent.com/27980260/90684603-1e1fe080-e22e-11ea-9c0c-3c2bb5f12583.png)
Hopefully, the screenshot loads and I am not making more of a fool out of myself :]
☺My app breaks at screen width  824-1130 and 481 - 620 which may not be an issue since somebody saved some suggested screen limits for us.
☺ I can't read the mental health post, not sure if we should put the title into a hover event, text-shadow, or gradient background cover.
☺ The card separation doesn't seem to be an issue unless one of the images breaks like this, just had a little time understanding what was going on in the page. Not sure if the designers would be okay having some color change in the cards or box-shadow.

But like I said runs great, looks good, and really sorry for spamming your pull request review. I am new to open git projects :]

JElgar commented:
> I like to nest my queries together at the end of the file also, but unless the leads 👍 this formatting idea change, we have to style it this way too.

@tunangst The reason I've been pushing this rather unorthodox approach is to keep the styling for individual components as contained as possible. This means if someone comes and works on a small section of the blog, eg update the tiles, they should be able to view all the style for that tile component in one place without jumping around the file. I do completely agree that in a normal case (where you develop in the mindset of pages rather than components)  that having the media queries all in once place is probably better. 

I'm very happy to change the style that we go for tho if people think having single media queries with all components nested in them is better? Or if we find out that this is a terrible system (as maybe people are mostly working on whole pages rather than editing components) then we can make the switch later on.

tunangst commented:
> 
> 
> > I like to nest my queries together at the end of the file also, but unless the leads 👍 this formatting idea change, we have to style it this way too.
> 
> @tunangst The reason I've been pushing this rather unorthodox approach is to keep the styling for individual components as contained as possible. This means if someone comes and works on a small section of the blog, eg update the tiles, they should be able to view all the style for that tile component in one place without jumping around the file. I do completely agree that in a normal case (where you develop in the mindset of pages rather than components) that having the media queries all in once place is probably better.
> 
> I'm very happy to change the style that we go for tho if people think having single media queries with all components nested in them is better? Or if we find out that this is a terrible system (as maybe people are mostly working on whole pages rather than editing components) then we can make the switch later on.

@JElgar Thank you for your explanation :]. It makes sense when you say it like that. This is my first experience in a team dev environment, so I do apologize if I say anything that sounds convoluted or strange.

JElgar commented:
@tunangst thank you so much for your review. I think to save time we are going to merge this in now buuut I completely agree with what you are saying and if @shah-tejas is happy we should work on that for the next release of the blog (with the search stuff).

shah-tejas commented:
Sure works for me as well! We can merge this for now, and make the other changes when implementing the remaining Blog features.
